docs/links/sdks/go.mdx

```diff
@@ -0,0 +1,10 @@
+---
+title: Go SDK
+url: https://github.com/modelcontextprotocol/go-sdk
+icon: go
```

Comment:
I don't know how to provide a file for this. (I can't find where the other languages' icons live.)